.github/workflows/continuous-integration-workflow.yml

```diff
@@ -0,0 +1,310 @@
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Cache Go vendor packages
+        uses: actions/cache@v1
+        with:
+          path: /go/pkg/mod
+          key: $${{ secrets.CACHE_VERSION }}-go-mod-$${ hashFiles('go.sum') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION }}-go-mod
+      - name: Download Go vendor packages
+        run: go mod download
+      - name: Set up Yarn for Contracts
+        run: yarn setup:contracts
+      - name: Run tests
+        run: go test -v -p 6 -parallel 4 ./...
+        env:
+          DATABASE_URL: postgres://postgres:postgres@postgres:5432/postgres?sslmode=disable
+
+  rust:
+    name: Rust
+    runs-on: ubuntu-latest
+    container:
+      image: smartcontract/builder:1.0.31
+      env:
+        CARGO_HOME: /root/.cargo
+        RUSTUP_HOME: /root/.rustup
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run Rust test
+        run: ./tools/ci/rust_test
+
+  geth-postgres:
+    name: Geth Postgres
+    runs-on: self-hosted
+    env:
+      GETH_MODE: true
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  parity-postgres:
+    name: Parity Postgres
+    runs-on: self-hosted
+    env:
+      CI: true
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Run tests
+        run: cd ./tools/docker && timeout --foreground 1200s ./compose test
+
+  solidity:
+    name: Solidity
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-yarn
+      - name: Install Yarn dependencies
+        run: make yarndep
+      - name: Cache Python dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: $${{ secrets.CACHE_VERSION}}-pip-$${ hashFiles('requirements.txt') }}
+          restore-keys: |
+            $${{ secrets.CACHE_VERSION}}-pip
+      - name: Install python packages
+        run: pip3 install -r requirements.txt
+      - name: Run tests
+        run: ./tools/ci/solidity_test
+
+  styleguide:
+    name: Styleguide
+    runs-on: ubuntu-latest
+    container: smartcontract/builder:1.0.31
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+      - name: Cache Yarn dependencies
+        uses: actions/cache@v1
+        with:
+          path: /usr/local/share/.cache/yarn
+          key: $${{ secrets.CACHE_VERSION}}-yarn-$${ hashFiles('yarn.lock') }}
+          restore-keys: |
```

Comment:
Any reason for the multi-line here?

Reply:
All the GH docs used multi-line for multiple restore-keys -- it's not necessary here but I thought if we changed the restore-key scheme it's a convenient reminder. Happy to drop 'em!

Reply:
Ah okay, we can just leave it in for having a minimal git diff in the future. Sound good to me